### PR TITLE
Bump node version to 16.13.0

### PR DESCRIPTION
**Pull request by gantunesr**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -491,7 +491,7 @@
     "fs": "react-native-level-fs"
   },
   "engines": {
-    "node": "^14.0.0",
+    "node": "^16.13.0",
```
Comment (sethkfman) [inline package.json:494]: I think we need to bump this in Bitrise as well. Look for the` nvm@1.3.0` step

Comment (gantunesr) [reply package.json:494]: Done in [528cc2e](https://github.com/MetaMask/metamask-mobile/pull/5934/commits/528cc2e45e7c7fcf9cf9148b67b69975422a9cb0)

**Comment by Gudahtt** [comment]
Just curious: was there a reason for choosing v16.13 specifically, rather than v16?

**Comment by Gudahtt** [comment]
Could you update the README as well? It references version 14 in the setup instructions

**Comment by gantunesr** [comment]
Hey @Gudahtt, I just now realized you left some comments,

> _Just curious: was there a reason for choosing v16.13 specifically, rather than v16?_

I think `v16` should work. The reason is that we need `v16.13` for the snaps system integration into mobile

> _Could you update the README as well? It references version 14 in the setup instructions_

Yes

**Comment by gantunesr** [comment]
Done @Gudahtt

**Review by Gudahtt — APPROVED** [review]
LGTM!

**Review by leotm — APPROVED** [review]
nice ^ we officially make the switch to `.node-version` and set v16 in [0.70](https://github.com/facebook/react-native/blob/0.70-stable/template/_node-version)
but our 0.66.5 `.nvmrc` ahead on 16's perfectly fine

but after SES/lockdown likely needs work LavaMoat-side to sandbox compile/run time
since we've been on [v14](https://github.com/LavaMoat/LavaMoat/commit/dc027eb3572e4f53b96cf5a1fd33b7bd700be88b) only couple months - i'll check w the team to plan/start early

i also bumped to [v18](https://github.com/facebook/react-native/pull/35443) fb picked for [0.71](https://github.com/facebook/react-native/blob/0.71-stable/template/_node-version)+ (but got few months till Node 16 EOL: 2023-09-11)
our 0.71.4 branch (i'm using for lockdown) is behind on [v16.14.0](https://github.com/MetaMask/metamask-mobile/blob/upgrade/react-native/0.71.4/.node-version) so also checking this is ok

Node 19's been around few months too (got unofficially working for RN)
so hopefully can bundle the LavaMoat work to support 16/18/19 at same time

**Pull request merged**